### PR TITLE
Delete app from tmpl on unsubscription callback

### PR DESCRIPTION
**Pull request by alextargov**
**Description**

Changes proposed in this pull request:
- ...

**Related issue(s)**

- ...

**Pull Request status**

- [ ] Implementation
- [ ] Unit tests
- [ ] Integration tests
- [ ] `chart/compass/values.yaml` is updated 
- [ ] Mocks are regenerated, using the automated script

### CODE REVIEW

File: components/director/internal/domain/apptemplate/repository.go [hunk 1]
```diff
@@ -77,6 +80,31 @@ func (r *repository) Get(ctx context.Context, id string) (*model.ApplicationTemp
 	return result, nil
 }
 
+// GetByFilters builds a subquery for a given slice of labelfilter.LabelFilter and gets a model.ApplicationTemplate
+func (r *repository) GetByFilters(ctx context.Context, filter []*labelfilter.LabelFilter) (*model.ApplicationTemplate, error) {
```
Comment (kaloyanspiridonov) [inline components/director/internal/domain/apptemplate/repository.go:84]: Add unit tests

File: components/director/internal/domain/apptemplate/service.go [hunk 2]
```diff
@@ -133,6 +135,16 @@ func (s *service) Get(ctx context.Context, id string) (*model.ApplicationTemplat
 	return appTemplate, nil
 }
 
+// GetByFilters gets a model.ApplicationTemplate by given slice of labelfilter.LabelFilter
```
Comment (kaloyanspiridonov) [inline components/director/internal/domain/apptemplate/service.go:138]: Add unit tests

File: tests/tenant-fetcher/tests/handler_test.go [hunk 3]
```diff
@@ -24,230 +24,338 @@ import (
...
-			// WHEN
-			addRegionalTenantExpectStatusCode(t, providedTenantIDs, http.StatusBadRequest)
-
-			// THEN
-			tenants, err := fixtures.GetTenants(certSecuredGraphQLClient)
-			require.NoError(t, err)
-			assert.Equal(t, oldTenantState.TotalCount, tenants.TotalCount)
+	t.Run("Application flows", func(t *testing.T) {
+		//GIVEN
+		tmplName := "app-flow-tmpl-name"
```
Comment (kaloyanspiridonov) [inline tests/tenant-fetcher/tests/handler_test.go:277]: use the following format for app template name "SAP <name> (<region>)"

**Review by kaloyanspiridonov — CHANGES_REQUESTED** [review]

**Review by kaloyanspiridonov — APPROVED** [review]

**Comment by PetarTodorovv** [comment]
how/why this is merged with lots of failing tests and not approved PR - from access/process(OWNERS/CODEOWNERS) point of view and from functionality/code point of view?
image

**Pull request merged**